app/api/resources/rapid_view.py

```diff
@@ -0,0 +1,64 @@
+from urllib.parse import parse_qs
+from fastapi import APIRouter, Request, HTTPException, Query
+from fastapi.responses import RedirectResponse
+import logging
+import re
+from core.logging import InterceptHandler
+from core.config import ENSEMBL_URL
+from api.utils.metadata import get_genome_id_from_assembly_accession_id
+from api.utils.rapid import format_assembly_accession
+
+logging.getLogger().handlers = [InterceptHandler()]
+
+router = APIRouter()
+
+
+# Resolve rapid urls
+@router.get("/{species_url_name}", name="Rapid Species Home")
+@router.get("/{species_url_name}/", name="Rapid Species Home")
+@router.get("/{species_url_name}/{subpath:path}", name="Rapid Species Home")
+async def resolve_species(
+    request: Request, species_url_name: str, subpath: str = "", r: str = Query(None)
+):
+    assembly_accession_id = format_assembly_accession(species_url_name)
+
+    genome_object = get_genome_id_from_assembly_accession_id(assembly_accession_id)
+
+    if genome_object:
+        genome_id = genome_object["genome_tag"] or genome_object["genome_uuid"]
+
+        # Extract specific parameters because Ensembl url uses ; instead of &
+        raw_query_string = request.scope["query_string"].decode()
+        query_string = raw_query_string.replace(";", "&")
+        query_params = parse_qs(query_string)
```

Comment:
According to the docs, `parse_qs` accepts a named `separator` parameter; so theoretically, we could just say `parse_qs(query_string, separator=";")`. I do not know whether this is better or worse than replacing the semicolons manually. I also do not know whether a url can have a combination of semicolons and ampersands.